relay-metrics/src/aggregation.rs

```diff
@@ -521,19 +522,32 @@ impl BucketValue {
         }
     }
 
-    /// Estimates the number of bytes needed to encode the bucket.
-    /// Note that this does not necessarily match the exact memory footprint of the bucket,
+    /// Estimates the number of bytes needed to encode the bucket value.
+    /// Note that this does not necessarily match the exact memory footprint of the value,
     /// because datastructures might have a memory overhead.
     ///
     /// This is very similar to [`BucketValue::relative_size`], which can possibly be removed.
     pub fn cost(&self) -> usize {
-        match self {
-            Self::Counter(_) => 8,
-            Self::Set(s) => 4 * s.len(),
-            Self::Gauge(_) => 5 * 8,
-            // Distribution values are stored as maps of (f64, u32) pairs
-            Self::Distribution(m) => 12 * m.internal_size(),
-        }
+        // We use private type aliases and constants here to make sure that a change in the implementation
+        // does not change the cost model without being noticed.
+        // Note that when the actual data type of these bucket values changes, we have to update
+        // the cost model and potentially increase the limits that are applied to it.
+        type SetType = u32;
+        /// [`BucketValue`] struct currently fits into 48 bytes:
+        const SELF_SIZE: usize = 48;
+        // Distribution values are stored as maps of (f64, u32) pairs:
+        const DIST_SIZE: usize = mem::size_of::<f64>() + mem::size_of::<u32>();
```

Comment:
The count already has a typedef if you look at the definition of `DistributionValue`.

```suggestion
        const DIST_SIZE: usize = mem::size_of::<f64>() + mem::size_of::<Count>();
```